From d7a5fb5176fc76d78b162755b34b1009f729f201 Mon Sep 17 00:00:00 2001
From: Logapriya P <quic_logapriy@quicinc.com>
Date: Tue, 2 Jan 2024 14:16:27 +0530
Subject: [PATCH] hostapd: Fix Intel AX210 driver DHCP failure

With the latest Intel driver, MLE information sent in Assocation
Response packet is not handled properly.

Fixed it by avoiding to fill the MLE info for Non-MLE supported clients.

Signed-off-by: Sriram R <quic_srirrama@quicinc.com>
Signed-off-by: Logapriya P <quic_logapriy@quicinc.com>
---
 src/ap/ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -5058,7 +5058,7 @@ rsnxe_done:
 
 #ifdef CONFIG_IEEE80211BE
 	if (hapd->iconf->ieee80211be && !hapd->conf->disable_11be) {
-		if (hapd->conf->mld_ap)
+		if (hapd->conf->mld_ap && sta && sta->mld_info.mld_sta)
 			p = hostapd_eid_eht_basic_ml(hapd, p, sta->mld_info.links, false);
 		p = hostapd_eid_eht_capab(hapd, p, IEEE80211_MODE_AP);
 		p = hostapd_eid_eht_operation(hapd, p, IEEE80211_MODE_AP);
